pandas/io/parsers.py

```diff
@@ -1662,14 +1662,22 @@ def __init__(self, src, **kwds):
             # GH 14671
             if (self.usecols_dtype == 'string' and
                     not set(usecols).issubset(self.orig_names)):
-                raise ValueError("Usecols do not match names.")
+                missing = [c for c in usecols if c not in self.orig_names]
+                raise ValueError(
+                    "Usecols do not match columns, "
+                    "columns expected but not found: {}".format(missing)
```

Comment:
Let's use keyword arguments.  Same for your other error messages.

Reply:
Also, I see a lot of duplicate code here.  Let's abstract into a method (you can just create a private method outside of the class).  That will make your life easier.